demo/midgard-sdk/src/tx-builder/user-events/deposit/types.ts

```diff
@@ -1,22 +1,39 @@
-import { Address, PolicyId, Data, UTxO } from "@lucid-evolution/lucid";
-import { POSIXTimeSchema } from "@/tx-builder/common.js";
+import { Data } from "@lucid-evolution/lucid";
+import { OutputReferenceSchema, POSIXTimeSchema } from "@/tx-builder/common.js";
 
-export const DepositDatumSchema = Data.Object({
-  data: Data.Object({
-    startTime: POSIXTimeSchema,
-    endTime: POSIXTimeSchema,
-  }),
+export const DepositEventSchema = Data.Object({
+    id: OutputReferenceSchema,
+    info: Data.Object({
+      l2_address: Data.Bytes(),
+      l2_datum: Data.Nullable(Data.Bytes())
+    })
+});
+
+export type DepositEvent = Data.Static<typeof DepositEventSchema>;
+export const DepositEvent = DepositEventSchema as unknown as DepositEvent;
+
+export const DepositInfoSchema = Data.Object({
+    l2_address: Data.Bytes(),
+    l2_datum: Data.Nullable(Data.Bytes())
 });
 
-export type Datum = Data.Static<typeof DepositDatumSchema>;
-export const Datum = DepositDatumSchema as unknown as Datum;
+export type DepositInfo = Data.Static<typeof DepositInfoSchema>;
+export const DepositInfo = DepositInfoSchema as unknown as DepositInfo;
+
+export const DepositDatumSchema = Data.Object({
+    event: DepositEventSchema,
+    inclusion_time: POSIXTimeSchema, // inclusion time is important , time range , 
+    })
+
+export type DepositDatum = Data.Static<typeof DepositDatumSchema>;
+export const DepositDatum = DepositDatumSchema as unknown as DepositDatum;
```

Comment:
The convention we have adopted is to rely primarily on file paths and namespaces to distinguish datums, redeemers, etc. So this should be named just `Datum`:
```suggestion
export const Datum = DatumSchema as unknown as Datum;
```